### PR TITLE
fix: fixed documents upload & kyb submit

### PR DESCRIPTION
**Pull request by chesterkmr**
### Description
- Fixed broken file upload
- fixed 500 when submitting last step

(Note: Those fixes are taken from token-auth branch)

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: Note the additional `_` for private.

```suggestion
  private async __persistFileUrlsToDocuments(
```

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: Use the type we have for documents. 🙂

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: Please use return types as an exception. Especially if it's `any`. 🙏🏼

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: `Promise.all` expects an array of promises, I understand it's probably needed but it's also likely misplaced.

```suggestion
      documents.reduce((filesList, document) => {
```

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: Wouldn't `await` be here if anything?

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: Does validation fail if `decision` is omitted?

File: services/workflows-service/src/workflow/workflow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/workflow/workflow.service.ts]: Appreciated. ❤️

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: Note `entity` may hold both `id` and `ballerineEntityId`. Is there a specific need for `endUserId` and for it to be somewhere specific?

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: On `TDocument` `pages` were omitted, let's use `DefaultContextSchema['documents']` until we decide on a single source of truth. With this change we can remove all the types below related to pages.

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: Is this property the only one required in the JSON schema?

File: services/workflows-service/src/workflow/workflow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/workflow/workflow.service.ts]: The error here should be logged out without this try/catch + console.log here. Should probably either remove this before merging or at least use our logger.

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: TypeScript suggests `ballerineFileId` may be `undefined`. We should consider handling that case.

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: As long as we return, throw an error, or use `!` over `ballerineFileId` being `undefined` types are being inferred here as `File | null` as expected.

```suggestion
            const file = filesById[page.ballerineFileId!] ?? null;
```

File: services/workflows-service/src/workflow/workflow.service.ts
Comment (chesterkmr) [reply services/workflows-service/src/workflow/workflow.service.ts]: It was leftover and should not be there

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (liorzam) [inline services/workflows-service/src/collection-flow/collection-flow.service.ts]: In case one of the files has failed, what is the expected behavior ?

In case we would like to be more resilient, i can suggest to use `Promise.allSettled`

`Promise.allSettled` will never reject, it will resolve once all Promises in the array have either rejected or resolved

File: services/workflows-service/src/providers/file/file-provider/local-file.service.ts
Comment (liorzam) [inline services/workflows-service/src/providers/file/file-provider/local-file.service.ts]: Its would be better to use `path.join` whenever you concatenate a path

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (Omri-Levy) [reply services/workflows-service/src/collection-flow/collection-flow.service.ts]: Let's add to tech debt a task or a chance to revisit instances of `Promise.all` and discuss how we should handle them. Handling will be in a separate PR.

Comment (chesterkmr) [reply services/workflows-service/src/collection-flow/collection-flow.service.ts]: Currently if any of them fail then 500 excepted,edge case in current scope.

File: services/workflows-service/src/collection-flow/collection-flow.service.ts
Comment (chesterkmr) [reply services/workflows-service/src/collection-flow/collection-flow.service.ts]: Code is currently tested,type update will force me to update code which eventually could lead to issues.
So lets move forward,this place doesnt looks like issue which affects anything in current scope

**Review by liorzam — APPROVED** [review]
You have an approval from my side,
However please wait for @Omri-Levy's approval as well

**Review by alonp99 — APPROVED** [review]

**Pull request merged**
